mempool/clist_mempool.go

```diff
@@ -193,9 +230,10 @@ func (mem *CListMempool) PreUpdate() {
 	}
 }
 
+// Size returns the total number of transactions in the mempool (that is, all lanes).
 // Safe for concurrent use by multiple goroutines.
 func (mem *CListMempool) Size() int {
-	return mem.txs.Len()
+	return int(mem.numTxs.Load())
```

Comment:
```suggestion
	numTxs := mem.numTxs.Load()
	if numTxs > math.MaxInt {
		panic(fmt.Sprintf("mempool has too many transactions (corrupted?); %v > %v ", numTxs, math.MaxInt))
	}
	return int(numTxs)
```
Can we add this for type-cast safety?